AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -174,6 +178,42 @@ + (void)setCustomProperties:(MSCustomProperties *)customProperties {
 }
 #endif
 
++ (void)setAuthTokenDelegate:(id<MSAuthTokenDelegate>)authTokenDelegate {
+  [[MSAppCenter sharedInstance] setAuthTokenDelegate:authTokenDelegate];
+}
+
+- (void)setAuthTokenDelegate:(id<MSAuthTokenDelegate>)authTokenDelegate {
+  MSAuthTokenContext *authTokenContext = [MSAuthTokenContext sharedInstance];
```

Comment:
does this logic need to be synchronized?

Reply:
The delegates are already run in a `@synchronized` block in `MSAuthTokenContext`, is there some other way that using this code could cause deadlocks?